backend/src/org/commcare/resources/model/ResourceTable.java

```diff
@@ -298,100 +404,69 @@ public void prepareResources(ResourceTable master, CommCareInstance instance)
                     }
                 }
 
-                // Vector<Reference> refs = explodeAllReferences(r, this, master);
-
-                boolean handled = false;
-                UnreliableSourceException theFailure = null;
+                findResourceLocationAndInstall(r, invalid, upgrade, instance, master);
 
-                // TODO: Possibly check if resource status is local and proceeding
-                // to skip this huge (although in reality like one step) chunk
-
-                for (ResourceLocation location : r.getLocations()) {
-                    if (handled) {
-                        break;
-                    }
-                    if (location.isRelative()) {
-                        for (Reference ref : explodeReferences(location, r, this, master)) {
-                            if (location.getAuthority() == Resource.RESOURCE_AUTHORITY_LOCAL && invalid.contains(ref)) {
-                            } else {
-                                try {
-                                    handled = installResource(r, location, ref, this, instance, upgrade);
-                                } catch (UnreliableSourceException use) {
-                                    theFailure = use;
-                                }
-                                if (handled) {
-                                    break;
-                                }
-                            }
-                        }
-                    } else {
-                        try {
-                            handled = installResource(r, location, ReferenceManager._().DeriveReference(location.getLocation()), this, instance, upgrade);
-                            if (handled) {
-                                break;
-                            }
-                        } catch (InvalidReferenceException ire) {
-                            ire.printStackTrace();
-                            // Continue until no resources can be found.
-                        } catch (UnreliableSourceException use) {
-                            theFailure = use;
-                        }
-                    }
-                }
-                if (!handled) {
-                    // If there wasn't a particular failure to point our finger at...
-                    if (theFailure == null) {
-                        throw new UnresolvedResourceException(r, "No external or local definition could be found for resource " + r.getResourceId());
-                    } else {
-                        // Otherwise, expose the lossy failure rather than the generic one
-                        throw theFailure;
-                    }
-                }
                 if (stateListener != null) {
                     stateListener.resourceStateUpdated(this);
                 }
+
+                v = getUnreadyResources();
             }
-            v = GetUnreadyResources();
```

Comment:
This can't be taken out if getUnreadyResources() changes due to code within the while-loop.